File: src/graph/swing/handler/CellTracker.java
/**
 * Copyright (c) 2008, Gaudenz Alder
 */
part of graph.swing.handler;

//import graph.swing.GraphComponent;

//import java.awt.Color;
//import java.awt.event.MouseEvent;
//import java.awt.event.MouseListener;
//import java.awt.event.MouseMotionListener;

/**
 * Event handler that highlights cells. Inherits from CellMarker.
 */
public class CellTracker extends CellMarker implements MouseListener,
		MouseMotionListener
{

	/**
	 * 
	 */
	private static final long serialVersionUID = 7372144804885125688L;

	/**
	 * Constructs an event handler that highlights cells.
	 */
	public CellTracker(GraphComponent graphComponent, Color color)
	{
		super(graphComponent, color);

		graphComponent.getGraphControl().addMouseListener(this);
		graphComponent.getGraphControl().addMouseMotionListener(this);
	}

	/**
	 * 
	 */
	public void destroy()
	{
		_graphComponent.getGraphControl().removeMouseListener(this);
		_graphComponent.getGraphControl().removeMouseMotionListener(this);
	}

	/*
	 * (non-Javadoc)
	 * @see java.awt.event.MouseListener#mouseClicked(java.awt.event.MouseEvent)
	 */
	public void mouseClicked(MouseEvent e)
	{
		// empty
	}

	/*
	 * (non-Javadoc)
	 * @see java.awt.event.MouseListener#mouseEntered(java.awt.event.MouseEvent)
	 */
	public void mouseEntered(MouseEvent e)
	{
		// empty
	}

	/*
	 * (non-Javadoc)
	 * @see java.awt.event.MouseListener#mouseExited(java.awt.event.MouseEvent)
	 */
	public void mouseExited(MouseEvent e)
	{
		// empty
	}

	/*
	 * (non-Javadoc)
	 * @see java.awt.event.MouseListener#mousePressed(java.awt.event.MouseEvent)
	 */
	public void mousePressed(MouseEvent e)
	{
		// empty
	}

	/*
	 * (non-Javadoc)
	 * @see java.awt.event.MouseListener#mouseReleased(java.awt.event.MouseEvent)
	 */
	public void mouseReleased(MouseEvent e)
	{
		reset();
	}

	/*
	 * (non-Javadoc)
	 * @see java.awt.event.MouseMotionListener#mouseDragged(java.awt.event.MouseEvent)
	 */
	public void mouseDragged(MouseEvent e)
	{
		// empty
	}

	/*
	 * (non-Javadoc)
	 * @see java.awt.event.MouseMotionListener#mouseMoved(java.awt.event.MouseEvent)
	 */
	public void mouseMoved(MouseEvent e)
	{
		process(e);
	}

}

File: lib/swing/handler/cell_tracker.dart
/**
 * Copyright (c) 2008, Gaudenz Alder
 */
part of graph.swing.handler;

//import java.awt.Color;
//import java.awt.event.MouseEvent;
//import java.awt.event.MouseListener;
//import java.awt.event.MouseMotionListener;

/**
 * Event handler that highlights cells. Inherits from CellMarker.
 */
class CellTracker extends CellMarker implements event.MouseUpHandler, event.MouseMoveHandler {//MouseListener, MouseMotionListener {

  event.HandlerRegistration _mouseUpRegistration;
  event.HandlerRegistration _mouseMoveRegistration;

  /**
   * Constructs an event handler that highlights cells.
   */
  CellTracker(GraphComponent graphComponent, awt.Color color) : super(graphComponent, color) {
    _mouseUpRegistration = graphComponent.getGraphControl().addMouseUpHandler(this);
    _mouseMoveRegistration = graphComponent.getGraphControl().addMouseMoveHandler(this);
  }

  void destroy() {
    if (_mouseUpRegistration != null) {
      _mouseUpRegistration.removeHandler();
    }
    if (_mouseMoveRegistration != null) {
      _mouseMoveRegistration.removeHandler();
    }
  }

  void onMouseUp(event.MouseEvent e) {
    reset();
  }

  void onMouseMove(event.MouseEvent e) {
    process(e);
  }

}
